util/fetch/src/client.rs

```diff
@@ -169,12 +170,42 @@ impl Drop for Client {
 }
 
 impl Client {
+	/// Create a new fetch client with a proxy.
+	pub fn new_with_proxy(https_proxy_url: Option<String>) -> Result<Self, Error> {
+		let (tx_start, rx_start) = std::sync::mpsc::sync_channel(1);
+		let (tx_proto, rx_proto) = mpsc::channel(64);
+
+		Client::background_thread(tx_start, rx_proto, https_proxy_url)?;
+
+		match rx_start.recv_timeout(Duration::from_secs(10)) {
```

Comment:
consider changing ```Duration::from_secs(10)``` to a constant instead with something like ```const MAX_FETCH_TIME: Duration = Duration::from_secs(10);```

Reply:
Got it!